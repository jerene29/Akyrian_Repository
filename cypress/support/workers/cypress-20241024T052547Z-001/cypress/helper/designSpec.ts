import { GetStudyRevisionListDocument } from '../../src/graphQL/generated/graphql';
import { mockUserDataAdmin } from '../../src/constant/testFixtures';

type SpecNumber = `S-${number}-${number}`;

type SpecInfo = {
  // e.g S-09-010
  spec: SpecNumber;
  // title e.g headerSection, combined description becomes S-09-010 headerSection, but optional
  title?: string;
};

type Options = {
  loginAs?: string;
  skipTest?: boolean;
};

function resolveDescription(description: SpecNumber | SpecInfo) {
  return typeof description === 'string' ? description : `${description.spec} ${description.title}`;
}

const aliasGetStudyRevisionListDocument =
  'name' in GetStudyRevisionListDocument.definitions[0]
    ? GetStudyRevisionListDocument.definitions[0].name?.value
    : 'GetStudyRevisionListDocument';
/*
example usage
designSpec('S-19-2', () => {

  describe('...', ...)

  it('checks something...', () => {...})
});
*/
export function designSpec(
  description: SpecNumber | SpecInfo,
  callback: () => void,
  options?: Options,
) {
  before(() => {
    cy.intercept('POST', '/graphql', (req) => {
      if (req.body.operationName === aliasGetStudyRevisionListDocument) {
        req.alias = req.body.operationName;
      }
    });
    cy.beforeSetup(mockUserDataAdmin);
    cy.clearLocalStorageSnapshot();
    cy.saveLocalStorage();
    cy.wait(`@${aliasGetStudyRevisionListDocument}`);
  });
  if (options?.skipTest) {
    describe.skip(resolveDescription(description), callback);
  } else if (typeof description !== 'string' && description.title) {
    describe(description.spec, () => {
      describe(String(description.title), callback);
    });
  } else {
    describe(resolveDescription(description), callback);
  }
}
