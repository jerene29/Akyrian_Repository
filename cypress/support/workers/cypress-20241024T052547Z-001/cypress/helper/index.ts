export * from './designSpec';
export * from './object';