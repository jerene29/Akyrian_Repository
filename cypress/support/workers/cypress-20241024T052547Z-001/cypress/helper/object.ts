/**
 * to shorten the tedious '[data-cy=selector]' selector.
 * just put the name of data-cy attributed element.
 * -- d stands for document
 *
```
d`addButton`
```
 */
export function d(args: TemplateStringsArray) {
  return `[data-cy=${args.join('')}]`;
}
