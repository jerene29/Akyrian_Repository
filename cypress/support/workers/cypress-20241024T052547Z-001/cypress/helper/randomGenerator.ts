import moment = require("moment");

export const randomAlphaNumeric = (length: number) => {
  const chars = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
  let result = '';
  for (let i = length; i > 0; --i) result += chars[Math.floor(Math.random() * chars.length)];
  return result;
}

export const randomAlphabet = (length: number) => {
  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
  let result = '';
  for (let i = length; i > 0; --i) result += chars[Math.floor(Math.random() * chars.length)];
  return result;
}

export const randomDate = () => {
  const start = new Date(2012, 0, 1)
  const end = new Date()
  const date = new Date(start.getTime() + Math.random() * (end.getTime() - start.getTime()))
  return moment(date);
}