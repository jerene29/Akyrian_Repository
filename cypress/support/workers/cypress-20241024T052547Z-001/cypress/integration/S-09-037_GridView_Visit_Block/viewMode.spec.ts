import { d, designSpec } from '../../helper';

const options = {
  loginAs: 'admin@example.com',
};

designSpec(
  'S-09-037',
  () => {
    before(() => {
      cy.createStudy({
        name: 'GRIDVIEW_TEST',
        formCount: 5,
        visitCount: 5,
      });
    });

    context('inactive edit mode', () => {
      it('checks if fields have a light blue border', () => {
        cy.hover(d`headerCell-0-2`);
        cy.getStyleSnapshot('borderColor', d`visitNameBox`);
        cy.shouldBeVisible(d`visitNameInactiveInput`);
      });

      it('checks if mouse pointer is a cursor icon over editable fields, clicking triggers active edit mode', () => {
        cy.get(d`visitNameBox`).click();
        cy.shouldBeVisible(d`visitNameInput`);
      });

      it('shows the drag handler', () => {
        cy.hover(d`headerCell-0-2`);
        cy.shouldBeVisible(d`moveHandler`);
      });

      it('reveals anchor checkbox', () => {
        cy.hover(d`headerCell-0-5`);
        cy.get(d`anchorCheckbox`).should('exist');
      });

      it('reveals delete button, and successfully delete visit', () => {
        cy.hover(d`headerCell-0-5`);
        cy.shouldBeVisible(d`deleteVisit`).click();
        cy.shouldBeVisible(d`confirmModal-confirmButton`).click();
        cy.isGone(d`headerCell-0-5`);
      });

      it('reveals + icons to the right and left of the block - clickable and can add a singular visit to the right', () => {
        cy.hover(d`headerCell-0-4`);
        cy.shouldBeVisible(d`copyVisitRightButton-0-4`).click();
        cy.shouldBeVisible(d`headerCell-0-5`);
        cy.shouldBeVisible(d`copyVisitLeftButton-0-4`).click();
        cy.get(d`headerCell-0-6`).should('exist');
      });

      // TODO: rework on this intermittently fail on cloud.
      xit('can drag visit', () => {
        cy.get(d`headerCell-0-2`).realHover();
        cy.dragAndDrop(d`headerCell-0-2`, d`headerCell-0-4`);

        cy.shouldBeVisible(d`confirmModal-confirmButton`).click();
        cy.reload();
        // one click is not enough!
        cy.get(d`grid-view`).click();
        cy.get(d`grid-view`).click();
        cy.get(d`grid-view`).click();

        cy.getTextSnapshot(d`headerCell-0-2`);
        cy.getTextSnapshot(d`headerCell-0-4`);
      });
    });
  },
  options,
);
