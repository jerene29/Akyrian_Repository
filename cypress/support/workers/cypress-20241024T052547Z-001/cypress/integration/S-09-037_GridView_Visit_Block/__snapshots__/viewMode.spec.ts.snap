exports[`S-09-037 > inactive edit mode > checks if fields have a light blue border #0`] =
{
  "borderColor": "rgb(49, 75, 158)"
};

exports[`S-09-037 > inactive edit mode > can drag visit #0`] =
{
  "snapshot": "Anchor Visitmovedelete-2+2new visit 31 Day after Anchor"
};

exports[`S-09-037 > inactive edit mode > can drag visit #1`] =
{
  "snapshot": "-2+2new visit 23 Days after Anchor"
};
